include/boost/url/decode.hpp

```diff
@@ -16,25 +16,82 @@
 #include <boost/url/decode_view.hpp>
 #include <boost/url/error_types.hpp>
 #include <boost/url/string_view.hpp>
+#include <boost/url/pct_string_view.hpp>
 #include <boost/url/detail/decode.hpp>
 #include <boost/url/grammar/all_chars.hpp>
 #include <boost/url/grammar/charset.hpp>
 
 namespace boost {
 namespace urls {
 
-/** Validate and decode a string view
+/** Decode a string view into a buffer.
+
+    This function applies percent-decoding to
+    the given percent-encoded string, by
+    converting escape sequences into their
+    character equivalent.
+
+    The output is written to the destination,
+    and will be truncated if there is
+    insufficient space.
+
+    @par Example
+    @code
+    char *dest = new char[MAX_LENGTH];
```

Comment:
MAX_LENGTH can't compile? is this javadoc tested?

Reply:
https://github.com/alandefreitas/url/blob/8e7e7f0da3b9a622a432a6d7bb450c57c1c5572f/test/unit/decode.cpp#L433-L437

Reply:
MAX_LENGTH is not in the code snippet